pkg/controller/atlasdeployment/advanced_deployment.go

```diff
@@ -108,14 +111,25 @@ func cleanupTheSpec(deployment *mdbv1.AdvancedDeploymentSpec) {
 // It will also prevent from setting ANY of (electable | analytics | readonly) specs
 //
 //	if region config has enabled compute autoscaling
-func handleAutoscaling(kubeDeployment *mdbv1.AdvancedDeploymentSpec) {
+func handleAutoscaling(ctx *workflow.Context, desiredDeployment *mdbv1.AdvancedDeploymentSpec, currentDeployment *mongodbatlas.AdvancedCluster) error {
 	isDiskAutoScaled := false
-	cleanupInstanceSize := func(s *mdbv1.Specs) {
+	syncInstanceSize := func(s *mdbv1.Specs, as *mdbv1.AdvancedAutoScalingSpec) error {
 		if s != nil {
-			s.InstanceSize = ""
+			size, err := normalizeInstanceSize(ctx, s.InstanceSize, as)
+			if err != nil {
+				return err
+			}
+
+			if size == currentDeployment.ReplicationSpecs[0].RegionConfigs[0].ElectableSpecs.InstanceSize {
```

Comment:
Potentially can wrap this into something like `hasInstanceSizeChanged()` just to make it more readable :)